src/NBitcoin/ProvenBlockHeader.cs

```diff
@@ -112,6 +113,44 @@ public override void ReadWrite(BitcoinStream stream)
             this.CoinstakeSize = stream.ProcessedBytes - baseByteSize - this.MerkleProofSize - this.SignatureSize;
         }
 
+        /// <inheritdoc />
+        public override uint256 GetHash()
+        {
+            uint256 hash = null;
+            uint256[] innerHashes = this.hashes;
+
+            if (innerHashes != null)
+                hash = innerHashes[0];
+
+            if (hash != null)
+                return hash;
+
+            if (this.version > 6)
+                hash = this.CalculateHash();
+            else
```

Comment:
This is not correct, you should send to HashX13 hash calculator the payload not the uint256.
What is this method trying to achieve hash of the PH?
Why not just call base.GetHash() infect you don't even need to override this method at all.